app/helpers/course_summaries_helper.rb

```diff
@@ -55,13 +55,10 @@ def get_table_json_data(current_user)
   end
 
   def course_information
-    rubric_max = RubricCriterion.group(:assessment_id).sum(:max_mark)
-    flexible_max = FlexibleCriterion.group(:assessment_id).sum(:max_mark)
-    checkbox_max = CheckboxCriterion.group(:assessment_id).sum(:max_mark)
-    @max_marks = Hash[Assignment.all.map do |a|
-      [a.id, rubric_max.fetch(a.id, 0) + flexible_max.fetch(a.id, 0) + checkbox_max.fetch(a.id, 0)]
-    end
-    ]
+    @max_marks = Hash[Assignment.all.joins(:criteria)
+                               .where('criteria.bonus': false)
+                               .group('assessments.id')
+                               .pluck('assessments.id', 'SUM(criteria.max_mark)')]
```

Comment:
Hm, I think if you do a `.group` followed by a `.sum`, you get a hash already (mapping id to the value being summed). I think this is exactly what you want in this case.